htdocs/commande/list.php

```diff
@@ -223,15 +227,16 @@
 //$arrayfields['anotherfield'] = array('type'=>'integer', 'label'=>'AnotherField', 'checked'=>1, 'enabled'=>1, 'position'=>90, 'csslist'=>'right');
 $arrayfields = dol_sort_array($arrayfields, 'position');
 
-if (!$user->hasRight('societe', 'client', 'voir')) {
-	$search_sale = $user->id;
```

Comment:
It seems we loose this protection when we are in default mode:
If MAIN_SEE_SUBORDINATES not set and use has permission to read all, tis was used to restrict list on $search_sale at line 1011 (in old code file)
So you should restore a restriction into the select. For example in your addition line 888
`if (!$permissiontoreadallthirdparty && getDolGlobalInt('MAIN_SEE_SUBORDINATES')) {`
you can do
```
if (!$permissiontoreadallthirdparty) {
 if (getDolGlobalInt('MAIN_SEE_SUBORDINATES')) {
      // restrict on list in userschilds
} else {
   // default mode, restrict on user->id  only.


```

Reply:
yes I see I did